wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -0,0 +1,242 @@
+	// Get top 20 topics ordered by count, then sort them alphabetically.
+	$topics = get_terms(
+		array(
+			'taxonomy' => 'topic',
+			'orderby' => 'count',
+			'order' => 'DESC',
+			'number' => 20,
+			'hide_empty' => true,
+		)
+	);
+	usort(
+		$topics,
+		function ( $a, $b ) {
+			return strcmp( strtolower( $a->name ), strtolower( $b->name ) );
+		}
+	);
+
+	$selected = isset( $wp_query->query['wporg_workshop_topic'] ) ? (array) $wp_query->query['wporg_workshop_topic'] : array();
+	$count = count( $selected );
+	$label = sprintf(
+		/* translators: The dropdown label for filtering, %s is the selected term count. */
+		_n( 'Topic <span>%s</span>', 'Topic <span>%s</span>', $count, 'wporg-learn' ),
+		$count
+	);
+
+	return array(
+		'label' => $label,
+		'title' => __( 'Topic', 'wporg-learn' ),
+		'key' => 'wporg_workshop_topic',
+		'action' => get_current_url(),
+		'options' => array_combine( wp_list_pluck( $topics, 'slug' ), wp_list_pluck( $topics, 'name' ) ),
+		'selected' => $selected,
+	);
+}
+
+
+/**
+ * Get the meta query values by key.
+ *
+ * @param WP_Query $query The query.
+ * @param string   $key The meta key.
+ * @return array   The meta query values.
+ */
+function get_meta_query_values_by_key( $query, $key ) {
+	if ( isset( $query->query_vars['meta_query'] ) ) {
+		$meta_query = $query->query_vars['meta_query'];
+
+		foreach ( $meta_query as $meta ) {
+			if ( isset( $meta['key'] ) && $meta['key'] === $key && ! empty( $meta['value'] ) ) {
+				return $meta['value'];
+			}
+		}
+	}
+
+	return array();
+}
+
+/**
+ * Get the list of languages for the course and lesson filters.
+ *
+ * @param array $options The options for this filter.
+ * @return array New list of language options.
+ */
+function get_language_options( $options ) {
+	global $wp_query;
+
+	$languages = get_available_post_type_locales( 'language', get_post_type(), 'publish' );
+	// if en_US is not in the list of languages, add it to the top
+	// as this is the default value for the meta field.
+	if ( ! isset( $languages['en_US'] ) ) {
+		$languages = array_merge( array( 'en_US' => 'English' ), $languages );
+	}
+
+	$selected = get_meta_query_values_by_key( $wp_query, 'language' );
+	$count = count( $selected );
+	$label = sprintf(
+		/* translators: The dropdown label for filtering, %s is the selected term count. */
+		_n( 'Language <span>%s</span>', 'Language <span>%s</span>', $count, 'wporg-learn' ),
+		$count
+	);
+
+	return array(
+		'label' => $label,
+		'title' => __( 'Language', 'wporg-learn' ),
+		'key' => 'language',
+		'action' => get_current_url(),
+		'options' => $languages,
+		'selected' => $selected,
+	);
+}
+
+/**
+ * Modify the query by adding meta query for language if set.
+ *
+ * @param WP_Query $query The query object.
+ */
+function modify_query( $query ) {
+	// Ensure this code runs only for the main query on archive pages
+	if ( ! is_admin() && $query->is_main_query() && $query->is_archive() ) {
+		$language = $_GET['language'];
```

Comment:
It shows this notice on my side, but doesn't when a language is selected.
![Screenshot 2024-06-07 at 02 11 39](https://github.com/WordPress/Learn/assets/18050944/24ada542-4cc7-47ad-b53d-461886775555)
But after clearing the selection, this screen shows up.
![image](https://github.com/WordPress/Learn/assets/18050944/44d238d0-dd11-4b1a-9bd1-5ebadeb6fc48)

I tried adding a check like this and both notices are gone.
```
if ( ! isset( $_GET['language'] ) ) {
    return;
}
```

Reply:
Fixed in [39ca990](https://github.com/WordPress/Learn/pull/2504/commits/39ca99020d07f8c6e7c093859ba5398db7ee78c2)